frontend/src/metabase/collections/containers/CollectionContent.jsx

```diff
@@ -41,15 +42,24 @@ function mapStateToProps(state) {
   };
 }
 
+const mapDispatchToProps = {
+  createBookmark: id => Bookmark.actions.create({ id: `collection-${id}` }),
+  deleteBookmark: id => Bookmark.actions.delete({ id: `collection-${id}` }),
+};
+
 function CollectionContent({
+  bookmarks,
   collection,
   collections: collectionList = [],
   collectionId,
+  createBookmark,
+  deleteBookmark,
   isAdmin,
   isRoot,
   handleToggleMobileSidebar,
   metadata,
 }) {
+  const [isBookmarked, setIsBookmarked] = useState(false);
```

Comment:
It seems that this can be changed to a selector, e.g. in `mapStateToProps`:

```
isBookmarked: bookmarks.some(bookmark => ... bookmark.item_id === collection.id)
```

You might want to create `selectors.ts` file in `metabase/collections` folder and extract this function there:

```
const isBookmarked = (state: State, { collection, bookmarks }: { collection: Collection, bookmarks: Bookmark[] }) => {
  return bookmarks.some....
}  
```

and then in `mapDispatchToProps`:

```
isBookmarked: isBookmarked(state, props)
```